SCons/Tool/MSCommon/common.py

```diff
@@ -342,6 +342,18 @@ def normalize_env(env, keys, force: bool=False):
     if sys32_wbem_dir not in normenv['PATH']:
         normenv['PATH'] = normenv['PATH'] + os.pathsep + sys32_wbem_dir
 
+    # ProgramFiles for PowerShell 7 Path and PSModulePath
+    progfiles_dir = os.environ.get("ProgramFiles")
+    if not progfiles_dir:
```

Comment:
Can `ProgramFiles` be undefined?

Reply:
Sure.

Could retrieve from registry first.

Reply:
I believe this code path might be invoked when configuring the default Environment on Windows with MSVC.

There is no guarantee that the user's OS environment is "sane" (e.g., redefined `COMSPEC`, variables missing, etc).  Such user environment issues can propagate a long way before hitting something objectionable.

Reply:
Gotcha. Ok. Was more a question of curiosity than a suggested code change.